src/parsers/messagesjson.js

```diff
@@ -0,0 +1,105 @@
+import RJSON from 'relaxed-json';
+
+import * as messages from 'messages';
+import JSONParser from 'parsers/json';
+import { MESSAGES_JSON, MESSAGE_PLACEHOLDER_REGEXP } from 'const';
+import { validateMessages } from 'schema/validator';
+import log from 'logger';
+
+export default class MessagesJSONParser extends JSONParser {
+  constructor(jsonString, collector, {
+    filename = MESSAGES_JSON, RelaxedJSON = RJSON,
+  } = {}) {
+    super(jsonString, collector, { filename });
+    this.relaxedJSON = RelaxedJSON;
+  }
+
+  parse() {
+    super.parse(this.relaxedJSON);
+
+    // Set up some defaults in case parsing fails.
+    if (typeof this.parsedJSON === 'undefined' || this.isValid === false) {
+      this.parsedJSON = {};
+    } else {
+      // We've parsed the JSON; now we can validate the manifest.
+      this._validate();
+    }
+  }
+
+  errorLookup(error) {
+    // This is the default message.
+    let baseObject = messages.JSON_INVALID;
+
+    // This is the default from webextension-manifest-schema, but it's not a
```

Comment:
Looking through the code I'm not even sure if this is used, as the additional properties are both custom error codes.

Reply:
I'm not sure I follow, which code do you mean and which additional properties?

Reply:
Ah, I think I agree, you're not re-using `error` and `error.message` unless I'm blind. So yes, that tidying isn't needed imho

Reply:
The code after this comment, which was copied over from the manifest parser (as the comment shows...)

Reply:
Ah yeah, as said before - I don't think we need it since `error` isn't re-used if I'm right.